prog/dftb+/lib_dftb/thirdorder.F90

```diff
@@ -565,6 +516,77 @@ subroutine addGradientDcXlbomd(this, neighList, species, coords, img2CentCell, q
 
   end subroutine addGradientDcXlbomd
 
+
+  !> Add stress component resulting from the derivative of the potential.
+  subroutine addStressDc(this, neighList, species, coords, img2CentCell, cellVol, st)
+
+    !> Instance.
+    class(ThirdOrder), intent(inout) :: this
+
+    !> Neighbor list.
+    type(TNeighborList), intent(in) :: neighList
+
+    !> Specie for each atom.
+    integer, intent(in) :: species(:)
+
+    !> Coordinate of each atom.
+    real(dp), intent(in) :: coords(:,:)
+
+    !> Mapping of atoms to cetnral cell.
+    integer, intent(in) :: img2CentCell(:)
+
+    !> cell volume.
+    real(dp), intent(in) :: cellVol
+
+    !> Gradient on exit.
+    real(dp), intent(inout) :: st(:,:)
+
+    integer :: iAt1, iAt2, iAt2f, iSp1, iSp2, iSh1, iSh2, iNeigh, ii
+    real(dp) :: rab, tmp, tmp3(3), stTmp(3,3), prefac, vect(3)
+    logical :: damping
+
+    stTmp = 0.0_dp
+    do iAt1 = 1, this%nAtoms
+      iSp1 = species(iAt1)
+      do iNeigh = 1, this%nNeighMax(iAt1)
+        iAt2 = neighList%iNeighbor(iNeigh, iAt1)
+        iAt2f = img2CentCell(iAt2)
+        iSp2 = species(iAt2f)
+        if (iNeigh > this%nNeigh(iSp2, iAt1)) then
+          cycle
+        end if
+        vect(:) = coords(:,iAt1) - coords(:,iAt2)
```

Comment:
I was wondering, whether we should refactor the part below it in a separate routine, as we call it here and also in the forceDC routine.

Reply:
Possibly not until the XLBOMD and H5 changes are made compatible.